pycbc/waveform/waveform.py

```diff
@@ -88,8 +88,21 @@ def _check_lal_pars(p):
         The lal type dictionary to pass to the lalsimulation waveform functions.
     """
     lal_pars = lal.CreateDict()
-    #nonGRparams can be straightforwardly added if needed, however they have to
-    # be invoked one by one
+    # try and add all provided items to lal_pars. We try both the provided key
+    # as well as a camel case version
+    for key, value in p.items():
+        if (value is None) or (key in default_args.keys()):
+            continue
+        camelcase = "".join([subkey.capitalize() for subkey in key.split("_")])
```

Comment:
Can any of the following lines be removed then if this is added? It would seem to serve in large part a similar purpose. Not all of course (modes for example would not work), but some of the nongr params probably could be not explicitly given now.

Reply:
@ahnitz, I would need to change the logic to account for the `nongr` params, or potentially deprecate the existing `pycbc` names and change them to new names. For example, the current `nongr` params is `pycbc` are follow the convention e.g. `dchi0` but the `lalsimulation` function requires them to be `NonGRDChi0`. I could remove `frame_axis` and `modes_choice`, but they would also need to be removed from the `default_args` (in order to prevent the issue [above](https://github.com/gwastro/pycbc/pull/4728#issuecomment-2088171686).

Reply:
No, I don't think we want to change the standardized names. One could remove code duplication with a dict, though, no?

Also, have you checked how slow this procedure is? What sort of overhead does it introduce?

Reply:
Well, the LALDict internally uses exactly the same field name as pycbc anyway (https://git.ligo.org/lscsoft/lalsuite/-/blob/master/lalsimulation/lib/LALSimInspiralWaveformParams.c#L1326) - it is only the name of the 'Insert' function that is unpredictable. 

After importing lalsimulation it is possible to get a mapping from the dictionary key to the Insert function in any case - LVK people can see here how to do that via `dir(lalsimulation)` (it's not anything that should be secret ...)
https://git.ligo.org/waveforms/new-waveforms-interface/-/issues/49#note_1028009